planner/core/planbuilder.go

```diff
@@ -1982,6 +1986,7 @@ func (b *PlanBuilder) buildSetValuesOfInsert(ctx context.Context, insert *ast.In
 			}
 			return ErrBadGeneratedColumn.GenWithStackByArgs(assign.Column.Name.O, tableInfo.Name.O)
 		}
+		b.curClause = fieldList
```

Comment:
What’s the problem without this line?

Reply:
```sql
insert t set unknown_column = 1;
```
and then server throw error: `[planner:1054]Unknown column 'd' in ''` instead of `[planner:1054]Unknown column 'd' in 'field list'`